metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -541,3 +542,100 @@ def _convert_value(param):
             "See the run in the UI at %s" % run_url,
             bold=True,
         )
+
+
+@argo_workflows.command(help="List all runs of the workflow on Argo Workflows.")
+@click.option(
+    "--pending",
+    default=False,
+    is_flag=True,
+    help="List all runs of the workflow in Pending state on " "Argo Workflows.",
+)
+@click.option(
+    "--running",
+    default=False,
+    is_flag=True,
+    help="List all runs of the workflow in Running state on " "Argo Workflows.",
+)
+@click.option(
+    "--succeeded",
+    default=False,
+    is_flag=True,
+    help="List all runs of the workflow in Succeeded state on " "Argo Workflows.",
+)
+@click.option(
+    "--failed",
+    default=False,
+    is_flag=True,
+    help="List all runs of the workflow in Failed state on " "Argo Workflows.",
+)
+@click.option(
+    "--error",
+    default=False,
+    is_flag=True,
+    help="List all runs of the workflow in Error state on " "Argo Workflows.",
+)
+@click.option(
+    "--unknown",
+    default=False,
+    is_flag=True,
+    help="List all runs of the workflow in Unknown state on " "Argo Workflows.",
+)
+@click.pass_obj
+def list_runs(obj, pending, running, succeeded, failed, error, unknown):
+    states = []
+    if pending:
+        states.append("Pending")
+    if running:
+        states.append("Running")
+    if succeeded:
+        states.append("Succeeded")
+    if failed:
+        states.append("Failed")
+    if error:
+        states.append("Error")
+    if unknown:
+        states.append("Unknown")
+
+    executions = ArgoWorkflows.list(obj.workflow_name, states)
+    found = False
+
+    def format_timestamp(timestamp):
+        TS_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+        return datetime.strptime(timestamp, TS_FORMAT)
+
+    for execution in executions:
+        found = True
+        obj.echo(
+            "*argo-{id}* "
+            "startedAt: '{startedAt}' "
+            "finishedAt: '{finishedAt}' "
+            "*{status}*".format(
+                id=execution["metadata"]["name"],
+                status=execution["status"]["phase"],
```

Comment:
If we club `failed` and `error` in one state (`failed`), it might help to still display `status` as just `failed`